src/client.ts

```diff
@@ -1,118 +1,87 @@
-    const isLowLevelData = (data: {}): boolean => {
-      if (data instanceof Object) {
-        const keys = Object.keys(data)
-        return JSON.stringify(['meta', 'content', 'page', 'xhrs', 'main']) ===
-          JSON.stringify(keys)
-      }
-      return false
-    }
-    return !isLowLevelData(data)
-  }
-
-  async url (params: { url: any }, retryCounter = 0): Promise<object | boolean> {
+  async url (params: UrlParams, retryCounter = 0): Promise<any> {
     debug.extend('url')(params.url)
-    let data: Object | boolean = false
-    if (typeof this.cache !== 'undefined') {
-      data = await this.cache.check(params.url)
-      if (data !== false) {
-        const cacheValid = Client.$isValidCache(data)
-        if (cacheValid) return data
-        await this.cache.remove(params.url)
-      }
-    }
-    // eslint-disable-next-line no-warning-comments
-    // TODO: refactor together with low-level into retry functions
-    try {
-      data = await this.lowLevelClient.url(params)
+    params.responseFormat = params.responseFormat ?? 'json'
 
-      if (data === null || typeof data === 'boolean') {
-        // check if an object was returned
-        debug.extend('url').extend('error')(`failed to execute url; data was null or boolean: retry count: ${retryCounter}`)
-        // throw error and let catch block handle whether to continue or not
-        throw new Error('Fetching response')
-      } else if (typeof this.cache !== 'undefined') {
-        // if an object was returned, save it to cache
-        await this.cache.write(params.url, data)
-        return data
-      }
+    try {
+      const data = await this.axios.get(`${this.uri}/url`, {params})
+        .then(response => response.data)
+      return data
     } catch (e) {
       // if we are within max retries allowed, call self recursively
       if (retryCounter < this.retries) {
         debug.extend('url').extend('error')(`retry/${retryCounter + 1}: ${e.message as string}`)
-        return await this.url(params, retryCounter + 1)
-      }
-
-      debug.extend('url').extend('error')(`retried ${this.retries} times; its not working`)
-      // remove the data from cache.
-      // it will be added when we can successfully get the data from authless-server
-      if (typeof this.lowLevelClient.cache !== 'undefined') {
-        await this.lowLevelClient.cache.remove(params.url)
+        return this.url(params, retryCounter + 1)
       }
 
       debug.extend('url').extend('error')(`retried ${this.retries} times; its not working`)
       // if we have extinguished max retries allowed, throw an error
       throw new VError(e, `failed to execute url; retried ${this.retries} times`)
     }
   }
+
+  async search (params: SearchParams): Promise<ResponseData> {
+    debug.extend('search')(params.q)
+    params.responseFormat = params.responseFormat ?? 'json'
+
+    return await this.axios.get(`${this.uri}/search`, {params})
+      .then(response => response.data)
+  }
 }
```

Comment:
Is `/search` a valid endpoint at the authless-server anymore? I don't think this is supported anymore, so we could remove this here.

Reply:
done